### ISSUE TITLE
Error decoding utf16/32 strings with a BOM

### ISSUE THREAD

**Issue by negasora** [Vector35/binaryninja-api#2880]
**Version and Platform (required):**
- Binary Ninja Version: 3.0.3222-dev (if version is stable, please also test the latest development build via the "Update Channel" option)
- OS: Linux
- OS Version: Yes

**Bug Description:**
We fail to take byte order markers into account when detecting utf16/32 encoded strings

**Steps To Reproduce:**
Please provide all steps required to reproduce the behavior:
1. Get the binary and bndb from [repro.zip](https://github.com/Vector35/binaryninja-api/files/9339311/repro.zip)
2. Open the bndb
3. All the strings from 0x4010-0x40a0 should render as "ABCDE"

**Expected Behavior:**
We recognize byte order markers and use them to decode the string correctly

**Comment by negasora** [comment]
Here's the source for the binary: 
```
#include "stdio.h"
#include "stdlib.h"
#include "stdint.h"


static uint8_t utf8_no_bom[6] = {0x41, 0x42, 0x43, 0x44, 0x45, 0};
static uint8_t utf8_bom[9] = {0xef, 0xbb, 0xbf, 0x41, 0x42, 0x43, 0x44, 0x45, 0};

static uint16_t utf16_no_bom[6] = {0x41, 0x42, 0x43, 0x44, 0x45, 0};
static uint16_t utf16_be[7] = {0xfeff, 0x41, 0x42, 0x43, 0x44, 0x45, 0};
static uint16_t utf16_le[7] = {0xfffe, 0x4100, 0x4200, 0x4300, 0x4400, 0x4500, 0};


static uint32_t utf32_no_bom[6] = {0x41, 0x42, 0x43, 0x44, 0x45, 0};
static uint32_t utf32_be[7] = {0xfeff, 0x41, 0x42, 0x43, 0x44, 0x45, 0};
static uint32_t utf32_le[7] = {0xfffe0000, 0x41000000, 0x42000000, 0x43000000, 0x44000000, 0x45000000, 0};


int main() {
    return 0;
}
```

**Comment by cxnder** [comment]
Resolved in builds >= 3.1.3657
